src/compiler.ts

```diff
@@ -9164,6 +9173,7 @@ export class Compiler extends DiagnosticEmitter {
         this.currentType,
         expression.operand,
         resolver.currentThisExpression,
+        Type.i32,
```

Comment:
@dcodeIO It looks like we should use proper index type also for `compileUnaryPostfixExpression` and `compileUnaryPrefixExpression`. Could you give me a hint as to the best way to derive this in these methods?